python/deltalake/writer.py

```diff
@@ -322,6 +313,34 @@ def default(self, obj: Any) -> Any:
         return json.JSONEncoder.default(self, obj)
 
 
+def try_get_table_and_table_uri(
+    table_or_uri: Union[str, Path, DeltaTable],
+    storage_options: Optional[Dict[str, str]] = None,
+) -> Tuple[Optional[DeltaTable], str]:
+    """Parses `table_or_uri` and returns `table` & `table_uri`.
+
+    Raises a ValueError if `table_or_uri` is not of type `str`, `Path` or `DeltaTable`
+    """
+    if not isinstance(table_or_uri, (str, Path, DeltaTable)):
+        raise ValueError("table_or_uri must be a str, Path or DeltaTable")
+
+    if isinstance(table_or_uri, str):
+        if "://" in table_or_uri:
+            table_uri = table_or_uri
+        else:
+            # Non-existant local paths are only accepted as fully-qualified URIs
+            table_uri = "file://" + str(Path(table_or_uri).absolute())
```

Comment:
by now we actually extended the path handling on the rust side to also support relative paths and scheme-less paths, and also create a directory if it does not yet exists.

https://github.com/delta-io/delta-rs/blob/67b512f601c85fb89839735164f9a1c818d58f5f/rust/src/builder.rs#L353-L376

Is it maybe a good idea to rely on the same logic on the python side as well, so we are always consistent in the supported paths?

Of course supporting `Path` objects will always be python territory :).

Reply:
Thanks for the pointer! That makes the python code a bit simpler